examples/binding-scala/js/src/main/scala/com/thoughtworks/todo/Main.scala

```diff
@@ -0,0 +1,148 @@
+package com.thoughtworks.todo
+
+import com.thoughtworks.binding.Binding._
+import com.thoughtworks.binding.dom
+import scala.collection.GenSeq
+import scala.scalajs.js.annotation.JSExport
+import org.scalajs.dom._
+import org.scalajs.dom.ext._
+import org.scalajs.dom.raw._
+import upickle.default._
+
+@JSExport object Main {
+
+  final class Todo(val title: String, val completed: Boolean)
+
+  object Todo {
+    def apply(title: String, completed: Boolean) = new Todo(title, completed)
+    def unapply(todo: Todo) = Option((todo.title, todo.completed))
+  }
+
+  final case class TodoList(text: String, hash: String, items: BindingSeq[Todo])
+
+  object Models {
+    def load() = LocalStorage("todos-binding.scala") match {
+      case None => Seq()
+      case Some(json) => read[Seq[Todo]](json)
+    }
+    def save(todos: Seq[Todo]) = LocalStorage("todos-binding.scala") = write(todos)
+
+    val allTodos = Vars[Todo](load(): _*)
+
+    val localStorageMountPoint = new MultiMountPoint(allTodos) {
+      override def set(newValue: Seq[Todo]) = save(newValue)
+      override def splice(oldSeq: Seq[Todo], from: Int, that: GenSeq[Todo], replaced: Int) = {
+        save(oldSeq.view(0, from) ++ that ++ oldSeq.view(from + replaced, oldSeq.length))
+      }
+    }
+    localStorageMountPoint.watch()
+
+    val editingTodo = Var[Option[Todo]](None)
+
+    val all = TodoList("All", "#/", allTodos)
+    val active = TodoList("Active", "#/active", for {todo <- allTodos if !todo.completed} yield todo)
+    val completed = TodoList("Completed", "#/completed", for {todo <- allTodos if todo.completed} yield todo)
+    val todoLists = Seq(all, active, completed)
+
+    def getCurrentTodoList = todoLists.find(_.hash == location.hash).getOrElse(all)
+    val currentTodoList = Var(getCurrentTodoList)
+    @dom val hashBinding = location.hash = currentTodoList.each.hash
+    hashBinding.watch()
+    window.onhashchange = { _: Any => currentTodoList := getCurrentTodoList }
+  }
+  import Models._
+
+  @dom def header = {
+    <header className="header">
+      <h1>todos</h1>
+      <input className="new-todo" autofocus={true} placeholder="What needs to be done?" onkeydown={event: KeyboardEvent =>
```

Comment:
Fixed in https://github.com/ThoughtWorksInc/todo/commit/66611166d33a661cdeb3eb3c887e64e3b0615ee2

BTW: @oyvindberg I really wonder how you learnt the `CurrentTargetReference` tricks.